lib/pdf_fill/filler.rb

```diff
@@ -164,5 +160,20 @@ def process_form(form_id, form_data, form_class, file_name_extension, fill_optio
 
       combine_extras(file_path, hash_converter.extras_generator)
     end
+
+    def make_hash_converter(form_id, form_class, merged_form_data, fill_options)
```

Comment:
This new method is responsible for deciding whether to instantiate an `ExtrasGenerator` or an `ExtrasGeneratorV2`.

It also helps keep `#process_form`'s line count low, to appease Rubocop.